Java/search/Fibonacci_search.java

```diff
@@ -0,0 +1,133 @@
+/*
+1) Find a Fibonacci number that is greater than or equal to the size of the array in which we are searching for the key.
+2) Compare the key with the predecessor of the Fibonacci number obtained in Step 1 and store it in index.
+3) If the key and array element at index are equal then the key is present at position index + 1.        
+4) If the key is less than array element at index, then search the left sub-tree to index.
+5) If the given key is greater than the array element at index, then search the right sub-tree to index.  
+6) If the key is not found, repeat the steps from Step 1 to Step 5 as long as index = 0, that is, Fibonacci number >= array_size. After each iteration the size of array array_size is reduced.
+                
+*/
+
+package Java;
+
+import java.util.*;
+
+public class FibonacciSearch {
+
+    // function to find the predecessor of the Fibonacci number that is greater than
+    // or equal to array_Size .
+    public static int getFibanociNumber(int array_Size) {
+
+        int fibk = 0;
+        int fibk2 = 0;
+        int fibk1 = 1;
+
+        if (array_Size == 0 || array_Size == 1) {
+            return 0;
+        }
+
+        while (fibk < array_Size) {
+            fibk = fibk1 + fibk2;
+            fibk2 = fibk1;
+            fibk1 = fibk;
+        }
+        return fibk2;
+    }
+
+    public static void main(String[] args) {
+
+        // Creating Scanner Object to read input from user
+        Scanner s = new Scanner(System.in);
+        int sorted_Array[] = new int[100];
+        int key;
+        int index;
+        int low;
+        int high;
+        int flag;
+        int location;
+        int array_Size;
+
```

Comment:
You can initialize it like that as well `int key, index ,low, high, flag, location, array_Size;` saves lots of lines